File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer>map=new HashMap<>();
        int moreneeded=0;
        for(int i=0;i<nums.length;i++){
            int num=nums[i];
            moreneeded=target-num;
            if(map.containsKey(moreneeded)){
                return new int[]{map.get(moreneeded),i};
            }
            else{
                map.put(num,i);
            }
        }
        return new int[]{-1,-1};
    }
}
// class Solution {
//     public int[] twoSum(int[] nums, int target) {
//             int n=nums.length;
//             int [][]arr=new int[n][2];
//             for(int i=0;i<nums.length;i++){
//                 arr[i][0]=nums[i];
//                 arr[i][1]=i;
//             }
//             Arrays.sort(arr,Comparator.comparingInt(a->a[0]));
//             int left=0,right=nums.length-1;
//             while(left<right){
//                 int sum=arr[left][0]+arr[right][0];
//                 if(sum==target){
//                     return new int[]{arr[left][1],arr[right][1]};
//                 }
//                 else if(sum<target){
//                     left++;
//                 }
//                 else{
//                     right--;
//                 }
//             }
//             return new int[]{-1,-1};
//         }
// }

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        arr=[(num,i) for i,num in enumerate(nums)]
        arr.sort()
        left,right=0,len(nums)-1
        sum=0
        while(left<right):
            sum=arr[left][0]+arr[right][0]
            if(sum==target):
                return [arr[left][1],arr[right][1]]
            elif(sum<target):
                left+=1
            else:
                right-=1


# class Solution:
#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#         mp={}
#         for i,num in enumerate(nums):
#             diff=target-num
#             if diff in mp:
#                 return [mp[diff],i]
#             mp[num]=i